lib/ApiRequestor.php

```diff
@@ -241,6 +245,43 @@ private static function _specificAPIError($rbody, $rcode, $rheaders, $resp, $err
         }
     }
 
+    /**
+     * @static
+     *
+     * @param string $rbody
+     * @param int    $rcode
+     * @param array  $rheaders
+     * @param array  $resp
+     * @param array  $errorData
+     *
+     * @return Exception\ApiErrorException
+     */
+    private static function _specificV2APIError($rbody, $rcode, $rheaders, $resp, $errorData)
+    {
+        $msg = isset($errorData['message']) ? $errorData['message'] : null;
+        $code = isset($errorData['code']) ? $errorData['code'] : null;
+        $type = isset($errorData['type']) ? $errorData['type'] : null;
+
+        switch ($type) {
+            case 'idempotency_error':
+                return Exception\IdempotencyException::factory($msg, $rcode, $rbody, $resp, $rheaders, $code);
+            // The beginning of the section generated from our OpenAPI spec
+            case 'temporary_session_expired':
+                return Exception\TemporarySessionExpiredException::factory(
+                    $msg,
+                    $rcode,
+                    $rbody,
+                    $resp,
+                    $rheaders,
+                    $code
+                );
+
+            // The end of the section generated from our OpenAPI spec
+            default:
+                return self::_specificV1APIError($rbody, $rcode, $rheaders, $resp, $errorData);
```

Comment:
Shouldnt we be checking for $code == "invalid_fields" here before falling back to v1 errors?